### PR TITLE
Code refactoring for ember-views package

### PR DESCRIPTION
**Pull request by artemgurzhii**

### CODE REVIEW

**Review by stefanpenner — APPROVED** [review]
🎉  👍

File: packages/ember-views/lib/system/event_dispatcher.js [hunk 1]
```diff
@@ -191,16 +189,18 @@ export default EmberObject.extend({
...
     @param {String} event the browser-originated event to listen to
     @param {String} eventName the name of the method to call on the view
-    @param {Object} viewRegistry
   */
-  setupHandler(rootElement, event, eventName, viewRegistry) {
+  setupHandler(rootElement, event, eventName) {
     let self = this;
 
+    let owner = getOwner(this);
+    let viewRegistry = owner && owner.lookup('-view-registry:main') || fallbackViewRegistry;
```
Comment (GavinJoyce) [inline packages/ember-views/lib/system/event_dispatcher.js]: @artemgurzhii was there a reason that you reverted [this change](https://github.com/emberjs/ember.js/pull/14784) in this PR?

Comment (GavinJoyce) [inline packages/ember-views/lib/system/event_dispatcher.js]: I've brought back that original change in this PR: https://github.com/emberjs/ember.js/pull/14968

Comment (artemgurzhii) [inline packages/ember-views/lib/system/event_dispatcher.js]: I don't think so, It look's like I have made mistake, sorry for that.

Comment (GavinJoyce) [inline packages/ember-views/lib/system/event_dispatcher.js]: :-) thanks

**Pull request merged**